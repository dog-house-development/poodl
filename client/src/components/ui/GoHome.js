import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import _ from 'lodash';

const propTypes = {
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    kind: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
    content: PropTypes.string,
    buttonOrLink: PropTypes.oneOf(['button', 'link'])
};

const defaultProps = {
    content: 'Home',
    size: 'medium',
    kind: 'primary'
};

export class GoHome extends Component {
    getClassName() {
        if (this.props.buttonOrLink) {
            return `${this.props.buttonOrLink} ${this.props.size} ${this.props.kind}`;
        }
    }

    render() {
        return (
            <Link to={this.props.isAuthenticated ? '/dashboard' : '/'} className={this.getClassName()}>
                {this.props.content}
            </Link>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        isAuthenticated: _.get(state.auth, 'isAuthenticated')
    };
};

GoHome.propTypes = propTypes;
GoHome.defaultProps = defaultProps;

export default connect(mapStateToProps)(GoHome);
