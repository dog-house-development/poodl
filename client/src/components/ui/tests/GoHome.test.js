import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import { GoHome, mapStateToProps } from '../GoHome';

configure({ adapter: new Adapter() });

describe('GoHome tests', () => {
    let wrapper, instance, state, props;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        state = _.assign(
            {},
            {
                auth: {
                    isAuthenticated: false
                }
            },
            _state
        );
        props = _.assign(
            {},
            {
                size: 'medium',
                kind: 'primary',
                content: 'Go Home'
            },
            _props
        );
        wrapper = shallow(<GoHome {..._.assign({}, props, mapStateToProps(state, props))} />);
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('render', () => {
        it('should render correctly when admin is authenticated', () => {
            const authenticatedState = {
                auth: {
                    isAuthenticated: true
                }
            };
            setInstanceAndWrapper({}, authenticatedState);
            expect(wrapper).toMatchSnapshot();
        });
        it('should render correctly when admin is not authenticated', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('mapStateToProps', () => {
        it('should map state to props', () => {
            expect(mapStateToProps(state, props)).toEqual({
                isAuthenticated: false
            });
        });
    });
});
