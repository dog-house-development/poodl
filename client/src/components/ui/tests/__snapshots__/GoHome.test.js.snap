// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoHome tests render should render correctly when admin is authenticated 1`] = `
<Link
  replace={false}
  to="/dashboard"
>
  Home
</Link>
`;

exports[`GoHome tests render should render correctly when admin is not authenticated 1`] = `
<Link
  replace={false}
  to="/"
>
  Home
</Link>
`;
